qchem/pennylane_qchem/qchem/structure.py

```diff
@@ -593,23 +595,25 @@ def _terms_to_qubit_operator(coeffs, ops, wires=None):
     q_op = QubitOperator()
     for coeff, op in zip(coeffs, ops):
 
-        # Pauli axis names, note s[-1] expects only 'Pauli{X,Y,Z}'
-        pauli_names = [s[-1] for s in op.name]
-
```

Comment:
Moved this part after the check with raising `ValueError`, as it would assume operations on `Pauli{X,Y,Z}` and Identity